### ISSUE TITLE
Provide support for automatic documentation of request and response structure using JSON schema

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-restdocs#43]
Spring Data REST can provide a JSON schema for each repository. Explore the possibility of using the returned schema to automatically generate documentation of request and response structure.

**Comment by baron1405** [comment]
+1

**Comment by dsebastien** [comment]
+1

**Comment by sdeleuze** [comment]
See also https://github.com/spring-projects/spring-hateoas/issues/444 related issue

**Comment by adamdec** [comment]
Hi, anyone thought how we could approach this one?

**Comment by adamdec** [comment]
ANyd I cna work on it if you let me and assign this ticket to me.

**Comment by wilkinsona** [comment]
@adamdec Thanks. I can't assign the issue to you due to GitHub's permissions model. Your comment here is sufficient though.

**Comment by adamdec** [comment]
As to make it KISS for the very beginning is it a good approach for you to read "schema.json" files (for Request/Response) from local directory like src/main/test/resources ?

**Comment by wilkinsona** [comment]
No, as I said in the opening description, the intention is to try to utilise the schema that's auto-generated and served by Spring Data REST
